tests/PregTest.php

```diff
@@ -159,4 +159,22 @@ public function testSplitNonUtf8Pattern()
 
         $this->assertSame($expectedResult, $actual);
     }
+
+    public function testCorrectnessForUtf8String()
+    {
+        Preg::match('/./', 'àbc', $methodMatches);
+        preg_match('/./', 'àbc', $functionMatches);
+
+        $this->assertSame(array('à'), $methodMatches);
+        $this->assertNotSame(array('à'), $functionMatches);
```

Comment:
here and in next, please extract expected argument into named var, to highly expose it's the same value for both assertions